ample/modelling/rosetta_model.py

```diff
@@ -741,23 +750,11 @@ def run_scripts(self, job_scripts, job_time=None, job_name=None, monitor=None):
                                         submit_max_array=self.submit_max_array)
 
     def setup_domain_restraints(self):
-        """
-        Create the file for restricting the domain termini and return the path to the file
-        """
-        logger.info('restricting termini distance: {0}'.format( self.domain_termini_distance ))
-        fas = open(self.fasta)
-        seq = ''
-        for line in fas:
-            if not re.search('>', line):
-                seq += line.rstrip('\n')
-        length = 0
-        for x in seq:
-            if re.search('\w', x):
-                length += 1
-                
-        restraints_file = os.path.join(self.work_dir, 'constraints')
-        with open(restraints_file, "w") as conin:
-            conin.write('AtomPair CA 1 CA {0} GAUSSIANFUNC {1} 5.0 TAG\n'.format(length,self.domain_termini_distance))
+        """Create the file for restricting the domain termini and return the path to the file."""
+        logger.info('restricting termini distance: {0}'.format(self.domain_termini_distance))
+        restraints_file = os.path.join(self.work_dir, 'domain_constraints.txt')
+        with open(restraints_file, "w") as w:
+            w.write('AtomPair CA 1 CA {0} GAUSSIANFUNC {1} 5.0 TAG\n'.format(self.sequence_length, self.domain_termini_distance))
```

Comment:
should be done in the restraints module. there is logic for handling all of this (I think)

Reply:
Where's that - I couldn't find it

Reply:
https://github.com/rigdenlab/ample/blob/53ccf3c48ea159a541c5c99f39947d42fdcb45ac/ample/modelling/energy_functions.py#L192

Reply:
Addressed with commit 7e5d35d